src/vivarium_cluster_tools/psimulate/worker/vivarium_work_horse.py

```diff
@@ -203,3 +223,54 @@ def format_and_record_details(
                 df.insert(df.shape[1] - 1, col_name, val)
         finished_results_metadata[key] = val
     return finished_results_metadata
+
+
+def run(sim: SimulationContext, job_parameters: JobParameters):
+    backup_freq = job_parameters.backup_configuration["backup_freq"]
+    if backup_freq:
+        backup_dir = job_parameters.backup_configuration["backup_dir"]
+        backup_path = (backup_dir / str(get_current_job().id)).with_suffix(".pkl")
+        time_to_save = time() + backup_freq
+        while not sim.past_stop_time():
+            sim.step()
+            if time() >= time_to_save:
+                sim.write_backup(backup_path)
+                time_to_save = time() + backup_freq
+    else:
+        sim.run()
+
+
+def get_backup(job_parameters: JobParameters) -> Optional[SimulationContext]:
+    backup_dir = job_parameters.backup_configuration["backup_dir"]
+    if not backup_dir.exists() or not os.listdir(backup_dir):
+        # No sim backups to load
+        return None
+    metadata_path = job_parameters.backup_configuration["backup_metadata_path"]
+    try:
+        pickle_metadata = pd.read_csv(metadata_path)
+    except FileNotFoundError:
+        return None
```

Comment:
Maybe a comment here. So this would mean that the sim _is_ creating backups and it's just that this specific job doesn't have one?

Reply:
Yep, either because it failed so early that it didnt save or somebody came along and deleted the backup.